binary_search_tree/binary_search_tree.py

```diff
@@ -30,7 +30,31 @@ def insert(self, value):
           complete = True
 
   def contains(self, target):
-    pass
+    current = self
+    complete = False
+
+    while not complete:
```

Comment:
you don't need `complete = False`. since you're breaking out of the loop in two conditions
`if not current` and `if current.value == target`
you can simply keep line 36 as `while True`
notice that you never ever change the value of `complete` after you set it on line 34.